internal/runners/ppm/convert.go

```diff
@@ -0,0 +1,191 @@
+package ppm
+
+import (
+	"github.com/ActiveState/cli/internal/analytics"
+	"github.com/ActiveState/cli/internal/constants"
+	"github.com/ActiveState/cli/internal/failures"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/internal/output"
+	"github.com/ActiveState/cli/internal/textutils"
+	"github.com/skratchdot/open-golang/open"
+)
+
+// convertAnswerCreate is the answer that the user can choose if they accept to create a virtual environment.  It is re-used at several places.
+var convertAnswerCreate string = locale.Tl("ppm_convert_answer_create", "Create Virtual Runtime Environment")
+
+// analyticsEventFunc is used to send analytics event
+type analyticsEventFunc func(string, string, string)
+
+// surveySelectFunc displays a menu with options that the user can select
+type surveySelectFunc func(message string, choices []string, defaultResponse string) (string, *failures.Failure)
+
+const (
+	askedWhy          string = "asked why"
+	seenStateToolInfo        = "visited state tool info"
+	seenPlatformInfo         = "visited platform info"
+	notConvinced             = "still wants ppm"
+)
+
+type conversionFlow struct {
+	survey    surveySelectFunc
+	out       output.Outputer
+	openURI   func(string) error
+	eventFunc analyticsEventFunc
+}
+
+// StartConversionFlowIfNecessary checks if the user is in a project directory.
+// If not, they are asked to create a project, and (in a wizard-kind-of way) informed about the consequences.
+func (p *Ppm) StartConversionFlowIfNecessary() error {
+	// start conversion flow only if we cannot find a project file
+	if p.project == nil {
+		cf := newConversionFlow(p.prompt.Select, p.out, open.Run, analytics.EventWithLabel)
+
+		analytics.Event(analytics.CatPpmConversion, "run")
+		r, err := cf.runSurvey()
+		analytics.EventWithLabel(analytics.CatPpmConversion, "completed", r.String())
```

Comment:
Please log an error state and a successful state, eg. http://github.com/ActiveState/cli/blob/156a8d6a5b4649a459888c2c7a99d43c2d40b67d/cmd/state/internal/cmdtree/tutorial.go#L51-L51